monero-sys/src/bridge.h

```diff
@@ -122,51 +122,115 @@ namespace Monero
         return wallet.createTransaction(dest_address, "", Monero::optional<uint64_t>(amount), 0, PendingTransaction::Priority_Default);
     }
 
+    /**
+     * Create a transaction that spends all the unlocked balance to a single destination.
+     */
     inline PendingTransaction *createSweepTransaction(
         Wallet &wallet,
         const std::string &dest_address)
     {
         return wallet.createTransaction(dest_address, "", Monero::optional<uint64_t>(), 0, PendingTransaction::Priority_Default);
     }
 
+    /**
+     * Creates a transaction that spends the unlocked balance to multiple destinations with given ratios.
+     * Ratiosn must sum to 1.
+     */
     inline PendingTransaction *createMultiSweepTransaction(
         Wallet &wallet,
         const std::vector<std::string> &dest_addresses,
-        const std::vector<double> &sweep_ratios)
+        const std::vector<double> &sweep_ratios) // Sweep ratios must sum to 1
     {
-        size_t N = dest_addresses.size();
-        if (N == 0 || sweep_ratios.size() != N)
+        size_t n = dest_addresses.size();
+
+        // Check if we have any destinations at all
+        if (n == 0)
+        {
+            // wallet.setStatusError("Number of destinations must be greater than 0");
             return nullptr;
+        }
 
-        uint64_t bal = wallet.unlockedBalance();
-        std::vector<uint64_t> amounts(N);
-        uint64_t sum = 0;
-        for (size_t i = 0; i + 1 < N; ++i)
+        // Check if the number of destinations and sweep ratios match
+        if (sweep_ratios.size() != n)
         {
-            amounts[i] = static_cast<uint64_t>(std::floor(bal * sweep_ratios[i]));
-            sum += amounts[i];
+            // wallet.setStatusError("Number of destinations and sweep ratios must match");
+            return nullptr;
         }
-        // reserve last slot for fee carve‐out
-        amounts[N - 1] = 0;
 
-        // build temporary dest list for fee estimation
+        // Check if the sweep ratios sum to 1
+        double sum_ratios = 0.0;
+        for (const auto &ratio : sweep_ratios)
+        {
+            sum_ratios += ratio;
+        }
+
+        const double epsilon = 1e-6;
+        if (std::abs(sum_ratios - 1.0) > epsilon)
+        {
+            // wallet.setStatusError("Sweep ratios must sum to 1.0");
+            return nullptr;
+        }
+
+        // To estimate the correct fee we build a transation that spends 1 piconero
```

Comment:
this will fail if we have less piconeros than addresses to spend to.

add a sanity check to fail if we have less piconeros than sweep addresses.